exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,213 @@
+    if (uniqueIdentifier != null) {
+      metricBuilder.putLabels(UNIQUE_IDENTIFIER_KEY, uniqueIdentifier);
+    }
+    return metricBuilder.build();
+  }
+
+  static MetricDescriptor mapMetricDescriptor(MetricData metric, String uniqueIdentifier) {
+    String instrumentName = metric.getInstrumentationLibraryInfo().getName();
+    MetricDescriptor.Builder builder = MetricDescriptor.newBuilder().setDisplayName(instrumentName)
+        .setType(DESCRIPTOR_TYPE_URL + instrumentName);
+    metric.getDescriptor().getConstantLabels().forEach((key, value) -> builder.addLabels(mapConstantLabel(key, value)));
+    if (uniqueIdentifier != null) {
+      builder.addLabels(
+          LabelDescriptor.newBuilder().setKey(UNIQUE_IDENTIFIER_KEY).setValueType(LabelDescriptor.ValueType.STRING)
+              .build());
+    }
+
+    MetricData.Descriptor.Type metricType = metric.getDescriptor().getType();
+    if (GAUGE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+    } else if (CUMULATIVE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
+    } else {
+      logger.log(WARNING, "Metric type {0} not supported", metricType);
+      return null;
+    }
+    if (LONG_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.INT64);
+    } else if (DOUBLE_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+    } else {
+      logger.log(WARNING, "Metric type {0} not supported", metricType);
+      return null;
+    }
+    return builder.build();
+  }
+
+  static MonitoredResource mapGcpMonitoredResource(Resource resource) {
+    ReadableAttributes attributes = resource.getAttributes();
+    if (attributes.get("cloud.provider") == null ||
+        !attributes.get("cloud.provider").getStringValue().equals("gcp")) {
+      return null;
+    }
+    String resourceType = attributes.get("gcp.resource_type").getStringValue();
+    if (!(resourceType.equalsIgnoreCase("gce_instance") || resourceType.equalsIgnoreCase("gke_container"))) {
+      return null;
+    }
+
+    MonitoredResource.Builder builder = MonitoredResource.newBuilder().setType(resourceType);
+    for (Map.Entry<String, String> labels : OTEL_TO_GCP_LABELS.get(resourceType).entrySet()) {
+      if (attributes.get(labels.getKey()) == null) {
+        logger.log(WARNING, "Missing monitored resource value for {0}", labels.getKey());
+        continue;
+      }
+      builder.putLabels(labels.getValue(), mapAttributeValueToString(attributes.get(labels.getKey())));
+    }
+    return builder.build();
+  }
+
+  static String mapAttributeValueToString(AttributeValue value) {
+    switch (value.getType()) {
+      case STRING:
+        return value.getStringValue();
+      case LONG:
+        return value.getLongValue() + "";
+      case DOUBLE:
+        return value.getDoubleValue() + "";
```

Comment:
I think we need to explicitly define the format for converting a double to string, otherwise we may get something like `12e7` (see https://docs.oracle.com/javase/6/docs/api/java/lang/Double.html#toString(double)), i.e. we need something like
```java
String.format("%f", double)
```
or you may look at other suggestions in https://stackoverflow.com/questions/16098046/how-do-i-print-a-double-value-without-scientific-notation-using-java